src/MultiFormats/NetworkProtocol.cs

```diff
@@ -0,0 +1,580 @@
+        {
+            var a = stream.ReadSomeBytes(_addressSize);
+            Address = new IPAddress(a);
+            Value = Address.ToString();
+        }
+    }
+
+    internal class Ipv6NetworkProtocol : IpNetworkProtocol
+    {
+        private static int _addressSize = IPAddress.IPv6Any.GetAddressBytes().Length;
+
+        public override string Name => "ip6";
+        public override uint Code => 41;
+
+        public override void ReadValue(TextReader stream)
+        {
+            base.ReadValue(stream);
+            if (Address.AddressFamily != AddressFamily.InterNetworkV6)
+                throw new FormatException(string.Format("'{0}' is not a valid IPv6 address.", Value));
+        }
+
+        public override void ReadValue(CodedInputStream stream)
+        {
+            var a = stream.ReadSomeBytes(_addressSize);
+            Address = new IPAddress(a);
+            Value = Address.ToString();
+        }
+    }
+
+    internal class P2PNetworkProtocol : NetworkProtocol
+    {
+        public MultiHash MultiHash { get; private set; }
+        public override string Name => "p2p";
+        public override uint Code => 421;
+
+        public override void ReadValue(TextReader stream)
+        {
+            base.ReadValue(stream);
+            MultiHash = new MultiHash(Value);
+        }
+
+        public override void ReadValue(CodedInputStream stream)
+        {
+            stream.ReadLength();
+            MultiHash = new MultiHash(stream);
+            Value = MultiHash.ToBase58();
+        }
+
+        public override void WriteValue(CodedOutputStream stream)
+        {
+            var bytes = MultiHash.ToArray();
+            stream.WriteLength(bytes.Length);
+            stream.WriteSomeBytes(bytes);
+        }
+    }
+
+    internal class IpfsNetworkProtocol : P2PNetworkProtocol
+    {
+        public override string Name => "ipfs";
+    }
+
+    internal class OnionNetworkProtocol : NetworkProtocol
+    {
+        public byte[] Address { get; private set; }
+        public ushort Port { get; private set; }
+        public override string Name => "onion";
+        public override uint Code => 444;
+
+        public override void ReadValue(TextReader stream)
+        {
+            base.ReadValue(stream);
+            var parts = Value.Split(':');
+            if (parts.Length != 2)
+                throw new FormatException(string.Format("'{0}' is not a valid onion address, missing the port number.",
+                    Value));
+            if (parts[0].Length != 16)
+                throw new FormatException(string.Format("'{0}' is not a valid onion address.", Value));
+            try
+            {
+                Port = ushort.Parse(parts[1]);
+            }
+            catch (Exception e)
+            {
+                throw new FormatException(
+                    string.Format("'{0}' is not a valid onion address, invalid port number.", Value), e);
+            }
+
+            if (Port < 1)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)